### PR TITLE
Bug 1589635: Treat linter warnings as errors in translate command.

### PR DESCRIPTION
**Pull request by mdboom**
Also breaks out the concept of glinter warnings and errors so we can
add checks that don't break the build in the future.

### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - `make test` runs without emitting any warnings
  - `make lint` runs without emitting any errors
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry to `HISTORY.rst` or an explanation of why it does not need one
  - Any breaking changes to language binding APIs are noted explicitly

### CODE REVIEW

File: HISTORY.rst [hunk 1]
```diff
@@ -5,6 +5,8 @@ History
 Unreleased
 ----------
 
+* **Breaking change:** glinter nits found during the `translate` command are now treated as errors, not warnings.
```
Comment (Dexterp37) [inline HISTORY.rst:8]: Let's also call out that glinter `errors` are treated as errors, not `warnings` (even though we don't have them at the moment). Are you planning on having a follow-up bug to discuss what's an error and what's a warning?

Comment (mdboom) [reply HISTORY.rst:8]: My thinking is that we would roll out new nits as warnings, and gradually move them to errors after users have had a chance to catch up.  It's possible on a case-by-case basis that some things would remain errors forever.

File: glean_parser/lint.py [hunk 2]
```diff
@@ -194,31 +194,50 @@ def check_misspelled_pings(
                 )
 
 
+# The checks that operate on an entire category of metrics:
+#    {NAME: (function, is_error)}
 CATEGORY_CHECKS = {
-    "COMMON_PREFIX": check_common_prefix,
-    "CATEGORY_GENERIC": check_category_generic,
-}  # type: Dict[str, Callable[[str, Iterable[metrics.Metric]], LintGenerator]]
+    "COMMON_PREFIX": (check_common_prefix, True),
```
[the remark this answers is not in the document]
Comment (mdboom) [reply glean_parser/lint.py:200]: That's a good idea.

File: HISTORY.rst (hunk 1 above)
Comment (Dexterp37) [reply HISTORY.rst:8]: That's fine! But in the long run, we're not going to break translate if we add new "warning classes", right?

Comment (mdboom) [reply HISTORY.rst:8]: That's right.  Warnings won't cause translate to return an error code from the process.

Comment (Dexterp37) [reply HISTORY.rst:8]: Cool, I'd just explicitly mention that in the changelog, for clarity for people outside of our group.

**Comment by mdboom** [comment]
I rebased this to include the new user lifetime expiration check, and made it a warning, so we can roll this out without breaking the build for folks.

**Pull request merged**